### PR TITLE
Change rechunk function parameter from default True, to default False

### PR DESCRIPTION
**Pull request by magnunor**
This pull request changes the default value for the `rechunk` parameter in all(?) functions from `True` to `False`. This to improve processing performance for large lazy datasets, as changing the chunking "structure" of large datasets can really be detrimental to performance.

For more information about this, see https://github.com/hyperspy/hyperspy/issues/3107

### Description of the change

- For all function parameters: change `rechunk=True`, to `rechunk=False`

### Progress of the PR
- [x] Change implemented (can be split into several points),
- [x] update docstring,
- [x] update user guide,
- [x] add an changelog entry in the `upcoming_changes` folder
- [x] Check formatting changelog entry in the `readthedocs` doc build of this PR (link in github checks)
- [x] add tests,
- [x] ready for review.

### Minimal example of the bug fix or the new feature

Make some test data
```python
import dask.array as da
import hyperspy.api as hs

data = da.arange(256*256*256*256).reshape(256, 256, 256, 256)
s = hs.signals.Signal2D(data).as_lazy()
s.save("testdata.zspy", chunks=(32, 32, 32 ,32))
```
Then run this for processing:
```python
s = hs.load("testdata.zspy", lazy=True)
s_sum_rechunk = s.sum(rechunk=True)
s_sum_not_rechunk = s.sum(rechunk=False)

s_sum_rechunk.compute() # 12.41 s
s_sum_not_rechunk.compute() # 3.7 s
```
I'm not really sure if this is the best example, as it might be a bit hardware dependent.

-----

There is one function which automatically does rechunking internally in the function, [integrate_simpson](https://github.com/hyperspy/hyperspy/blob/RELEASE_next_major/hyperspy/_signals/lazy.py#L706). This one should probably be changed as well, but since it didn't involve the `rechunk` parameter in its function call, I didn't change it in this pull request.

----------

@CSSFrancis

### CODE REVIEW

**Comment by CSSFrancis** [comment]
@magnunor This is great!  `Dask` is getting smarter and smarter but it hasn't quite figured out how to minimize the transfer between different workers and still makes some less than intelligent rechunking in certain cases.  

It would be a good idea to keep an eye out because sometimes `dask` will drop something like [2022.11.0](https://www.coiled.io/blog/reducing-dask-memory-usage) which is truly wonderful.

File: doc/user_guide/big_data.rst
Comment (CSSFrancis) [inline doc/user_guide/big_data.rst]: I think that this is confusing because the version added is `.. versionadded:: 1.3.2`

Maybe add in a new block 

```suggestion
.. versionadded:: 2.0.0

Starting in version 2.0.0 HyperSpy does not automatically rechunk datasets as this leads for reduced performance.  The ``rechunk`` or ``optimize`` keyword argument can be set to ``True`` to change the chunking which could potentially speed up operations.
```

**Review by CSSFrancis — COMMENTED** [review]
This looks good to me.  One point is that in the guide it might be nice to add somewhere that the default changed with the 2.0.0 version.  Otherwise this is good to merge.

**Comment by magnunor** [comment]
>  I think that this is confusing because the version added is .. versionadded:: 1.3.2
>
> Maybe add in a new block

Done!

I'll be away the whole of July, so if this PR is blocking the 2.0.0 release, feel free to improve and/or merge it. Maintainers should have access to change it.

File: doc/user_guide/big_data.rst
Comment (ericpre) [reply doc/user_guide/big_data.rst]: Would it make sense to keep both?

**Comment by ericpre** [comment]
> There is one function which automatically does rechunking internally in the function, [integrate_simpson](https://github.com/hyperspy/hyperspy/blob/RELEASE_next_major/hyperspy/_signals/lazy.py#L706). This one should probably be changed as well, but since it didn't involve the rechunk parameter in its function call, I didn't change it in this pull request.

I added the `rechunk` argument to `integrate_simpson` to enforce consistency.

**Review by ericpre — APPROVED** [review]

**Review by CSSFrancis — APPROVED** [review]
This looks good to me if we want to merge the changes.  It might be work rebasing the the branch before merging.

**Pull request merged**
